Add SSO configuration support to helm chart

flo-mic opened the pull request:
As SSO to Vaultwarden using OpenID Connect will be soon available via the PR https://github.com/dani-garcia/vaultwarden/pull/3899, it would be nice to control this with the recommended helm chart of vaultwarden. Therefore this PR adds the newly added parameters to the chart and enables SSO configuration.

The PR is already combining the settings from the original PR as well as the additions which @Timshell is nicely maintaining here https://github.com/Timshel/vaultwarden until the initial PR was merged.

guerzon commented:
Hi @flo-mic, first of all, thanks for the PR.

I was waiting for https://github.com/dani-garcia/vaultwarden/pull/3899 to be merged first but as of today it's not yet merged. I do not see harm in merging this PR but it could cause confusion/inconsistency when the values are already available and the chart documentation discusses the SSO feature but in reality the latest available image still does not support said feature. With that, I would wait until the feature is merged.

As a review point: please bump the chart version to `0.32.0`. Of course when the PR gets merged, the app version in Chart.yaml (and ideally `image.tag` in values.yaml) has to be updated as well.

Also: if you don't mind, can you run `./generate-readme.sh` (requires Docker) as well?

Thanks

flo-mic commented:
Hi @guerzon I agree to wait with the merge of this PR to avoid confusions. Also I have updated the chart version and run the generate-readme script.

Timshel commented:
I'll mention that all the `sso.roles` and `sso.organizations` config/features are not part of the merged PR.

flo-mic commented:
Hey, appologize the delay. I updated the PR and fixed the linting issues. Regarding the support for `sso.roles` and `sso.organizations` should I drop this as it was not part of the initial PR from Timshel?

flo-mic commented:
Sorry for the delay on this from my end. Should be fully up to date and be ready to be merged.
